tests/test_ext_autodoc.py

```diff
@@ -1679,6 +1679,41 @@ def test_final(app):
     ]
 
 
+@pytest.mark.sphinx('html', testroot='ext-autodoc')
+def test_overload(app):
+    options = {"members": None}
+    actual = do_autodoc(app, 'module', 'target.overload', options)
+    assert list(actual) == [
+        '',
+        '.. py:module:: target.overload',
+        '',
+        '',
+        '.. py:class:: Math',
+        '   :module: target.overload',
+        '',
+        '   docstring',
+        '',
+        '',
+        '   .. py:method:: Math.sum(x, y)',
+        '                  Math.sum(x: int, y: int) -> int',
+        '                  Math.sum(x: float, y: float) -> float',
+        '                  Math.sum(x: str, y: str) -> str',
```

Comment:
I'd argue the desired output is
```suggestion
        '   .. py:method:: Math.sum(x: int, y: int) -> int',
        '                  Math.sum(x: float, y: float) -> float',
        '                  Math.sum(x: str, y: str) -> str',
```

Reply:
We can give type annotations to the main function.
```
@overload
def sum(x: int, y: int) -> int:
   ...

@overload
def sum(x: str, y: str) -> str:
   ...

def sum(x: Any, y: Any) -> Any:
    return x + y
```

I think showing `(x: Any, y: Any) -> Any` is good. What do you think?

Reply:
The example for [`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) does not annotate the main function.

Reply:
Ah, [mypy does](https://mypy.readthedocs.io/en/stable/more_types.html), and says that

> It may or may not have type hints. If it does, mypy
> will check the body of the implementation against the
> type hints.
>
> Mypy will also check and make sure the signature is
> consistent with the provided variants.

This sounds like an implementation helper to me, and not something that is actually interesting to a user.

I'd therefore lean towards either:

1. Not showing the signature, as I suggest above
2. Providing an option (default off) to show the main signature

Probably easier to do 1, and wait to see if anyone asks for 2 :)

Reply:
Acceptable. I'll change to 1.